packages/cli/src/license/license.controller.ts

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable no-param-reassign */
+
+import express from 'express';
+import { LoggerProxy } from 'n8n-workflow';
+
+import { getLogger } from '@/Logger';
+import {
+	ILicensePostResponse,
+	ILicenseReadResponse,
+	InternalHooksManager,
+	ResponseHelper,
+} from '..';
+import { LicenseService } from './License.service';
+import { getLicense } from '@/License';
+import { LicenseRequest } from '@/requests';
+import { isInstanceOwner } from '@/PublicApi/v1/handlers/users/users.service';
+
+export const licenseController = express.Router();
+
+/**
+ * Initialize Logger if needed
+ */
+licenseController.use((req, res, next) => {
+	try {
+		LoggerProxy.getInstance();
+	} catch (error) {
+		LoggerProxy.init(getLogger());
+	}
+	next();
+});
+
+// Helper for getting the basic license data that we want to return
+async function getLicenseData(): Promise<ILicenseReadResponse> {
```

Comment:
@freyamade I'd recommend moving this function to the license service as well

Reply:
ooh sure thing!